### PR TITLE
Fix the appening deprecating warning being fatal

### LINKED ISSUE #3601: Deprecation warning is a hard failure
**Describe the bug**
The deprecation error for multi-app configs should be a warning, not a full error.

**Steps to Reproduce**
Run `waypoint up` with >1 app in a waypoint.hcl

**Expected behavior**
The warning should allow current multi-app behavior to continue.

* Waypoint CLI Version: 0.9.0

**Additional context**
[Discuss conversation](https://discuss.hashicorp.com/t/deprecating-projects-or-how-i-learned-to-love-apps/40888/8)

### PR DESCRIPTION
**Pull request by evanphx**
Fixes #3601

### CODE REVIEW

File: internal/config/validate.go [hunk 1]
```diff
@@ -72,7 +72,17 @@ func (v ValidationResults) Error() string {
...
+	for _, vr := range v {
+		if vr.Error != nil {
+			return true
+		}
+	}
+
+	return false
+}
+
+const AppeningHappening = "[DEPRECATION] More than one app stanza within a waypoint.hcl file is deprecated, and will be removed in a future version.\nPlease see https://discuss.hashicorp.com/t/deprecating-projects-or-how-i-learned-to-love-apps/40888 for more information."
```
Comment (izaaklauer) [inline internal/config/validate.go:85]:
```suggestion
const AppeningHappening = "[NOTICE] More than one app stanza within a waypoint.hcl file is under consideration for change or removal in a future version.\nTo give feedback, visit https://discuss.hashicorp.com/t/deprecating-projects-or-how-i-learned-to-love-apps/40888"
```

Comment (catsby) [inline internal/config/validate.go]: > [DEPRECATION] More than one app stanza within a waypoint.hcl file is deprecated, and will be removed in a future version

I thought we were still debating/researching this, perhaps we should be softer here and mention it's something we're thinking about but not 100% committed either way?

**Review by izaaklauer — CHANGES_REQUESTED** [review]
Looks good! Returning ValidationResults separately makes sense to me. One suggestion to update to the new softer wording as discussed with @briancain, but happy to discuss it.

File: internal/config/validate.go [hunk 2]
```diff
@@ -72,7 +72,17 @@ func (v ValidationResults) Error() string {
 	return fmt.Sprintf("%d validation errors: %s", len(v), strings.Join(values, ", "))
 }
 
-const AppeningHappening = "[DEPRECATION] More than one app stanza within a waypoint.hcl file is deprecated, and will be removed in 0.10.\nPlease see https://discuss.hashicorp.com/t/deprecating-projects-or-how-i-learned-to-love-apps/40888 for more information."
+func (v ValidationResults) Errors() bool {
```
Comment (catsby) [inline internal/config/validate.go:75]: `Errors()`->`HasErrors()`? 

A method `Errors()` makes me think I'm going to get a slice of errors returned, if any exist. I think `HasErrors()` matches the intent here better, similar to the HCL `HasErrors()` in other parts of this file

**Review by izaaklauer — APPROVED** [review]

**Review by catsby — APPROVED** [review]
~General comment on the strong wording that we will remove multi-apps things.~ nvm it was updated while I was typing that comment lol

Other nit about `Errors()` vs. `HasErrors()` but overall LGTM

File: internal/config/validate.go (hunk 1 above)
Comment (catsby) [reply internal/config/validate.go]: nvm!

File: internal/config/validate.go (hunk 2 above)
Comment (evanphx) [reply internal/config/validate.go:75]: Sure, good call. Changing now.

File: .changelog/3605.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```release-note:fix
```
Comment (briancain) [inline .changelog/3605.txt:1]: Should this be `bug`? I can't remember if go-changelog accepts arbitrary release note categories or if it's fixed to a certain set.

**Review by briancain — APPROVED** [review]
Looks good! And fixes up the hard failure for me in my testing 👍🏻

File: .changelog/3605.txt (hunk 3 above)
Comment (evanphx) [reply .changelog/3605.txt]: You're correct, fixed!

**Pull request merged**